store/sqlstore/supplier.go

```diff
@@ -176,9 +178,12 @@ func NewSqlSupplier(settings model.SqlSettings, metrics einterfaces.MetricsInter
 	supplier.stores.productNotices = newSqlProductNoticesStore(supplier)
 	err := supplier.GetMaster().CreateTablesIfNotExists()
 	if err != nil {
-		mlog.Critical("Error creating database tables.", mlog.Err(err))
-		time.Sleep(time.Second)
-		os.Exit(EXIT_CREATE_TABLE)
+		if dupError, ok := errors.Cause(err).(*pq.Error); ok && dupError.Code.Name() == DUP_ENTRY_ERROR {
```

Comment:
This would only work with Postgres I believe. Can we modify it to support [MySQL](https://dev.mysql.com/doc/mysql-errors/5.7/en/server-error-reference.html#error_er_table_exists_error) too?

It would also be great if we can refactor this entire thing to a function to just check for the error and return whether it's a duplicate error or not. That way we can also write a test for this.

Take a look at https://github.com/mattermost/mattermost-server/blob/master/store/retrylayer/retrylayer.go#L365-L379 for an example of how we check for repeatable errors coming from the database.

It's also more robust to use `errors.As` rather than to just typecast in case the error was wrapped.

Reply:
Superb, thanks for the context @agnivade. Pretty new to the codebase so i wasn't aware that this code path is exercised with multiple storages. Will amend to handle MySQL as well.

Reply:
amended in https://github.com/mattermost/mattermost-server/pull/16270/commits/4d980e8dc6900157fdf5ce8d62491d9bf88d60cb